aws-cpp-sdk-identity-management/source/auth/STSProfileCredentialsProvider.cpp

```diff
@@ -323,9 +323,17 @@ void STSProfileCredentialsProvider::Reload()
 AWSCredentials STSProfileCredentialsProvider::GetCredentialsFromSTSInternal(const Aws::String& roleArn, Aws::STS::STSClient* client)
 {
     using namespace Aws::STS::Model;
+
+    const Aws::String roleSessionName = [](){
+        Aws::StringStream ss;
+        ss << "aws-sdk-cpp-" << Aws::Utils::DateTime::CurrentTimeMillis();
+        return ss.str();
+    }();
+
     AssumeRoleRequest assumeRoleRequest;
     assumeRoleRequest
         .WithRoleArn(roleArn)
+        .WithRoleSessionName(roleSessionName)
```

Comment:
`#include <aws/core/utils/UUID.h>`
Then you can simply replace this with

`.WithRoleSessionName(Aws::Utils::UUID::RandomUUID())`

And we actually prefer UUID as the session name.

Reply:
Done! :-)